lib/services/android-project-service.ts

```diff
@@ -190,7 +190,11 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 			// will replace applicationId in app/App_Resources/Android/app.gradle if it has not been edited by the user
 			let userAppGradleFilePath = path.join(this.$projectData.appResourcesDirectoryPath, this.$devicePlatformsConstants.Android, "app.gradle");
-			shell.sed('-i', /__PACKAGE__/, this.$projectData.projectId, userAppGradleFilePath);
+			try {
+				shell.sed('-i', /__PACKAGE__/, this.$projectData.projectId, userAppGradleFilePath);
+			} catch(e) {
+				this.$errors.failWithoutHelp(` Error: ${e}.\nIt's possible you're using an old template. You can try updating it and try again.`);
```

Comment:
My question is - is there something really important in the default app.gradle that requires having it in all projects? Will the project work without the default app.gradle. In case "yes", this should be warning, not error.
Also what will happen in case I have app.gradle, but I do not have `__PACKAGE__` in it - the `sed` will probably fail - should this stop the execution?